src/libstore/globals.cc

```diff
@@ -83,6 +83,8 @@ void loadConfFile()
        ~/.nix/nix.conf or the command line. */
     globalConfig.resetOverridden();
 
+    settings.environment.overridden = settings.environment.hasInherited;
```

Comment:
Added a comment above this line. Basically if we get an element from the current environment we should consider it to be set dynamically so that when we contact the daemon we do send this over, even though we have the same config file as the daemon. Supports the `environment = https_proxy` use case.

This might be a bad idea, but this is the simplest thing that should work that I can think of.